packages/projen/component/dir-env/src/dirEnv.ts

```diff
@@ -0,0 +1,259 @@
+import { Component, FileBase, type Project, SourceCode } from 'projen'
+
+export enum DirEnvLayout {
+	/**
+	 * NodeJS.
+	 */
+	NODE = 'node',
+	/**
+	 * Python.
+	 */
+	PYTHON = 'python',
+}
+
+export enum DirEnvUse {
+	/**
+	 * ASDF.
+	 */
+	ASDF = 'asdf',
+	/**
+	 * RTX.
+	 */
+	RTX = 'rtx',
+}
+
+/**
+ * DirEnv commands available in .envrc context.
+ * @see https://github.com/direnv/direnv/blob/master/stdlib.sh
+ */
+export enum DirEnvStdLibCommand {
+	USE = 'use',
+	LAYOUT = 'layout',
+	STRICT_ENV = 'strict_env',
+	UNSTRICT_ENV = 'unstrict_env',
+	DIRENV_VERSION = 'direnv_version',
+	SOURCE_ENV = 'source_env',
+	SOURCE_ENV_IF_EXISTS = 'source_env_if_exists',
+	LOG_STATUS = 'log_status',
+	LOG_ERROR = 'log_error',
+	JOIN_ARGS = 'join_args',
+	EXPAND_PATH = 'expand_path',
+	DOTENV = 'dotenv',
+	DOTENV_IF_EXISTS = 'dotenv_if_exists',
+	ENV_VARS_REQUIRED = 'env_vars_required',
+}
+
+export enum DirEnvLogType {
+	INFO = DirEnvStdLibCommand.LOG_STATUS,
+	ERROR = DirEnvStdLibCommand.LOG_ERROR,
+}
+
+export interface DirEnvOptions {
+	/**
+	 * File name.
+	 */
+	readonly fileName?: string
+}
+
+export class DirEnv extends Component {
+	public static of(project: Project): DirEnv | undefined {
+		const isDirEnv = (c: Component): c is DirEnv => c instanceof DirEnv
+		return project.components.find(isDirEnv)
+	}
+
+	public readonly file: SourceCode
+
+	constructor(project: Project, public readonly options: DirEnvOptions = {}) {
+		super(project)
+		const { fileName = '.envrc' } = options
+		this.#validateFileName(fileName)
+		this.file = new SourceCode(project, fileName, {})
```

Comment:
```suggestion
		this.file = new SourceCode(project, fileName, { marker: true, read-only: true,})
```

Reply:
This might not work. `SourceCode` doesn't extend `FileBase` component

Reply:
:thinking: 
https://github.com/projen/projen/blob/7d1af8e1fa1794aeb8cf548c3b82a3e90e1e43cc/src/source-code.ts#L43-L45

Reply:
For some reason, when I jumped to `SourceCode` definition, it showed me 0.0.0 version of `SourceCode` without that marker getter. Sorry for confusion